pkgs/development/python-modules/pydantic/default.nix

```diff
@@ -1,138 +1,73 @@
-{ lib
-, stdenv
-, buildPythonPackage
-, autoflake
-, cython
-, devtools
-, email-validator
-, fetchFromGitHub
-, fetchpatch
-, pytest-mock
-, pytestCheckHook
-, python-dotenv
-, pythonAtLeast
-, pythonOlder
-, pyupgrade
-, typing-extensions
-# dependencies for building documentation.
-# docs fail to build in Darwin sandbox: https://github.com/samuelcolvin/pydantic/issues/4245
-, withDocs ? (stdenv.hostPlatform == stdenv.buildPlatform && !stdenv.isDarwin && pythonAtLeast "3.10")
-, ansi2html
-, markdown-include
-, mike
-, mkdocs
-, mkdocs-exclude
-, mkdocs-material
-, mdx-truly-sane-lists
-, sqlalchemy
-, ujson
-, orjson
-, hypothesis
-, libxcrypt
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  callPackage,
+  hatch-fancy-pypi-readme,
+  hatchling,
+  annotated-types,
+  pydantic-core,
+  typing-extensions,
+  email-validator,
+  pythonOlder,
+  pytestCheckHook,
+  dirty-equals,
+  pytest-examples,
+  pytest-mock,
+  faker,
 }:
-
 buildPythonPackage rec {
   pname = "pydantic";
-  version = "1.10.9";
-  format = "setuptools";
-
-  outputs = [
-    "out"
-  ] ++ lib.optionals withDocs [
-    "doc"
-  ];
-
-  disabled = pythonOlder "3.7";
+  version = "2.0.3";
+  format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "pydantic";
-    repo = pname;
-    rev = "refs/tags/v${version}";
-    hash = "sha256-POqMxBJUFFS1TnO9h5W7jYwFlukBOng0zbtq4kzmMB4=";
+    repo = "pydantic";
+    rev = "v${version}";
+    hash = "sha256-Nx6Jmx9UqpvG3gMWOarVG6Luxgmt3ToUbmDbGQTHQto=";
   };
 
   patches = [
-    # Fixes racy doctests build failures on really fast machines
-    # FIXME: remove after next release
-    (fetchpatch {
-      url = "https://github.com/pydantic/pydantic/pull/6103/commits/f05014a30340e608155683aaca17d275f93a0380.diff";
-      hash = "sha256-sr47hpl37SSFFbK+/h3hGlF6Pl6L8XPKDU0lZZV7Vzs=";
-    })
-  ];
-
-  postPatch = ''
-    sed -i '/flake8/ d' Makefile
-  '';
-
-  buildInputs = lib.optionals (pythonOlder "3.9") [
-    libxcrypt
+    ./01-remove-benchmark-flags.patch
   ];
 
   nativeBuildInputs = [
-    cython
-  ] ++ lib.optionals withDocs [
-    # dependencies for building documentation
-    autoflake
-    ansi2html
-    markdown-include
-    mdx-truly-sane-lists
-    mike
-    mkdocs
-    mkdocs-exclude
-    mkdocs-material
-    sqlalchemy
-    ujson
-    orjson
-    hypothesis
+    hatch-fancy-pypi-readme
+    hatchling
   ];
 
   propagatedBuildInputs = [
-    devtools
-    pyupgrade
+    annotated-types
+    pydantic-core
     typing-extensions
   ];
 
   passthru.optional-dependencies = {
-    dotenv = [
-      python-dotenv
-    ];
     email = [
       email-validator
     ];
   };
 
+  passthru.v1 = callPackage ./v1.nix {};
```

Comment:
That looks odd, is this actually used? I think we should remove it.

Reply:
That's an oversight, removed in https://github.com/NixOS/nixpkgs/compare/0734be7a465c129a4ff92b1179ca1c8dcbf027a9..27b5cc26162f07368235823ea6f37df7627fd945 :smirk: